src/main/java/org/kitteh/irc/client/library/implementation/ClientBuilder.java

```diff
@@ -221,6 +223,15 @@ public ClientBuilder webircRemove() {
         return this;
     }
 
+    @Nonnull
+    @Override
+    public ClientBuilder stsStorageManager(@Nonnull StsStorageManager storageManager) {
+        // Note, this will only catch users using the supplied insecure manager (per the docs).
+        Sanity.truthiness(!(this.config.get(Config.SSL_TRUST_MANAGER_FACTORY) instanceof AcceptingTrustManagerFactory), "Cannot use insecure trust manager factory with STS.");
```

Comment:
In that case, might be worth only checking `truthiness` if `storageManager` is not `null` to allow removing the storage manager after a trust manager factory is set.